src/DIRAC/Core/Utilities/Extensions.py

```diff
@@ -0,0 +1,215 @@
+"""TODO"""
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+import argparse
+from collections import defaultdict
+import fnmatch
+import functools
+import glob
+import importlib
+import os
+import pkgutil
+import sys
+
+import importlib_resources
+import six
+import DIRAC
+
+try:
+  from importlib.machinery import PathFinder
+except ImportError:
+  # Fallback for Python 2
```

Comment:
Just as reminder: we have several cases of using `imp` in DIRAC code, that one day we should address.

Reply:
I've already prepared the commits for v7r3 as part of fixing Python 3 and guiding the functions included in this module.